src/content/app-architecture/case-study/ui-layer.md

```diff
@@ -0,0 +1,652 @@
+This figure shows from a high-level how new data in the repository
+propagates up to the UI layer and triggers a re-build of your Flutter widgets.
+    </figcaption>
+</figure>
+
+1. New state is provided to the ViewModel from a Repository
+2. The ViewModel updates its UI state to reflect the new data
+3. ViewModel.notifyListeners is called, alerting the View of new UI State
+4. The View (widget) re-renders.
+
+For example, when the user navigates to the Home screen and the ViewModel is
+created, the `_load` method is called. Until this method completes, the UI state
+is empty, the view displays a loading indicator. When the `_load` method
+completes, if it is successful, there is new data in the ViewModel, and it must
+notify the view that new data is available.
+
+```dart title=home_viewmodel.dart highlightLines=19
+class HomeViewModel extends ChangeNotifier {
+  // ...
+
+ Future<Result> _load() async {
+    try {
+      final userResult = await _userRepository.getUser();
+      switch (userResult) {
+        case Ok<User>():
+          _user = userResult.value;
+          _log.fine('Loaded user');
+        case Error<User>():
+          _log.warning('Failed to load user', userResult.error);
+      }
+
+      // Load saved trips (omitted for brevity)
+
+      return userResult;
+    } finally {
+      notifyListeners();
+    }
+  }
+}
+
+```
+
+:::note
+`ChangeNotifier` and [`ListenableBuilder`][] (discussed later on this page) are 
+part of the Flutter SDK, 
+and provide a good solution for updating the UI when state changes. 
+You can also use a robust third-party state management solution, 
+such as [Riverpod][], [flutter_bloc][], or [Signals][]. 
+These libraries offer different tools for handling UI updates. 
+Read more about using `ChangeNotifier` in 
+our [state-management documentation][].
+:::
+
+## Define a View
+
+A [View][] is a widget within your app. 
+Often, a view represents one screen in your app that 
+has its own route and includes a [`Scaffold`][] at the top of the
+widget subtree, such as the `HomeScreen`, but this isn't always the case. 
+
+Sometimes a view is a single UI element that
+encapsulates functionality that needs to be re-used throughout the app. 
+For example, the Compass app has a view called `LogoutButton`, 
+which can be dropped anywhere in the widget tree that a user might 
+expect to find a logout button.
+The `LogoutButton` view has its own ViewModel called `LogoutViewModel`. 
+And on larger screens, there might be multiple views on screen that 
+would take up the full screen on mobile.
+
+:::note
+"View" is an abstract term, and one View does not equal one widget. 
+Widgets are composable, and several can be combined to create one view. 
+Therefore, ViewModels don't have a 1-to-1 relationship with widgets, 
+but rather a 1-to-1 relation with a *collection* of widgets.
+:::
+
+The widgets within a view have three responsibilities:
+
+* Display the data properties from the ViewModel
```

Comment:
Similar to earlier, list items should have consistent ending punctuation if possible.